pkg/monitor/monitor/monitor_manager.go

```diff
@@ -272,3 +285,44 @@ func (mm *MonitorManager) syncTidbMonitorRbac(monitor *v1alpha1.TidbMonitor) (*c
 
 	return sa, nil
 }
+
+func (mm *MonitorManager) syncIngress(monitor *v1alpha1.TidbMonitor) error {
+	if err := mm.syncPrometheusIngress(monitor); err != nil {
+		return err
+	}
+
+	return mm.syncGrafanaIngress(monitor)
+}
+
+func (mm *MonitorManager) syncPrometheusIngress(monitor *v1alpha1.TidbMonitor) error {
+	if monitor.Spec.Prometheus.Ingress == nil {
+		return mm.removeIngressIfExist(monitor, prometheusName(monitor))
+	}
+
+	ingress := getPrometheusIngress(monitor)
+	_, err := mm.typedControl.CreateOrUpdateIngress(monitor, ingress)
+	return err
+}
+
+func (mm *MonitorManager) syncGrafanaIngress(monitor *v1alpha1.TidbMonitor) error {
+	if monitor.Spec.Grafana == nil {
+		return mm.removeIngressIfExist(monitor, grafanaName(monitor))
+	} else if monitor.Spec.Grafana.Ingress == nil {
+		return mm.removeIngressIfExist(monitor, grafanaName(monitor))
+	}
+	ingress := getGrafanaIngress(monitor)
+	_, err := mm.typedControl.CreateOrUpdateIngress(monitor, ingress)
+	return err
+}
+
+// If the dashboard or dashboard ingress is nil when ingress also existed, we would remove Ingress
```

Comment:
```suggestion
// removeIngressIfExist removes Ingress if it exists
```

you didn't check any spec is nil in this function